oshdb-util/src/main/java/org/heigit/bigspatialdata/oshdb/util/time/ISODateTimeParser.java

```diff
@@ -31,60 +31,58 @@
    *  
    * @param isoDateTimeString ISO Date or ISO DateTime string
    * @return ZonedDateTime
-   * @throws Exception
+   * @throws Exception if Date or Date-Time pattern is not supported
    */
-  public static ZonedDateTime parseISODateTime(String isoDateTimeString) throws Exception{
+  public static ZonedDateTime parseISODateTime(String isoDateTimeString) throws Exception {
     ZonedDateTime zdt = null;
+
+    // always remove trailing Z, no other timezones are allowed anyways
+    if (isoDateTimeString.endsWith("Z")) {
+      isoDateTimeString = isoDateTimeString.substring(0,isoDateTimeString.length() - 1);
+    }
+
     int length = isoDateTimeString.length();
-    
+
     switch (length) {
-    case 4:
-      // pattern = "uuuu";
-      zdt = Year.parse(isoDateTimeString).atDay(1).atStartOfDay(ZoneId.of("Z"));
-      break;
-    case 6:
-      // pattern = "uuuuMM";
-      zdt = LocalDate.parse(isoDateTimeString+"01", DateTimeFormatter.BASIC_ISO_DATE).atStartOfDay(ZoneId.of("Z"));
-      break;
-    case 7:
-      // pattern = "uuuu-MM";
-      zdt = YearMonth.parse(isoDateTimeString).atDay(1).atStartOfDay(ZoneId.of("Z"));
-      break;
-    case 8:
-      // pattern = "uuuuMMDD";
-      zdt = LocalDate.parse(isoDateTimeString, DateTimeFormatter.BASIC_ISO_DATE).atStartOfDay(ZoneId.of("Z"));;
-      break;
-    case 10:
-      // pattern = "uuuu-MM-DD";
-      zdt = LocalDate.parse(isoDateTimeString, DateTimeFormatter.ISO_DATE).atStartOfDay(ZoneId.of("Z"));
-      break;
-      //with time
-    case 13:
-      // pattern = "uuuu-MM-DDTHH";
-      zdt = LocalDateTime.parse(isoDateTimeString+":00:00Z", DateTimeFormatter.ISO_DATE_TIME).atZone(ZoneId.of("Z"));
-      break;
-    case 16:
-      // pattern = "uuuu-MM-DDTHH:MM";
-      zdt = LocalDateTime.parse(isoDateTimeString+":00Z", DateTimeFormatter.ISO_DATE_TIME).atZone(ZoneId.of("Z"));
-      break;
-    case 19:
-      // pattern = "uuuu-MM-DDTHH:MM:ss";
-      zdt = LocalDateTime.parse(isoDateTimeString+"Z", DateTimeFormatter.ISO_DATE_TIME).atZone(ZoneId.of("Z"));
-      break;
-    case 20:
-      // pattern = "uuuu-MM-DDTHH:MM:ssZ";
-      zdt = ZonedDateTime.parse(isoDateTimeString, DateTimeFormatter.ISO_DATE_TIME);
-      break;
-    case 23:
-      // pattern = "uuuu-MM-DDTHH:MM:ss.sss";
-      zdt = LocalDateTime.parse(isoDateTimeString+"Z", DateTimeFormatter.ISO_DATE_TIME).atZone(ZoneId.of("Z"));
-      break;
-    case 24:
-      // pattern = "uuuu-MM-DDTHH:MM:ss.sssZ";
-      zdt = ZonedDateTime.parse(isoDateTimeString, DateTimeFormatter.ISO_DATE_TIME);
-      break;
-    default:
-      throw new Exception("Date or DateTime Format not supported: " + isoDateTimeString);
+      case 4:
+        // pattern = "uuuu";
+        zdt = Year.parse(isoDateTimeString).atDay(1).atStartOfDay(ZoneId.of("Z"));
```

Comment:
it could make sense to store `ZoneId.of("Z")` in a constant, because it is needed in each one of the swtich-cases here… maybe it is just a matter of taste, though